numba/cuda/cudaimpl.py

```diff
@@ -935,6 +938,28 @@ def ptx_atomic_cas_tuple(context, builder, sig, args):
                         'with %s array' % dtype)
 
 
+@lower(stubs.atomic.cas, types.Array, types.intp, types.Any, types.Any)
+@lower(stubs.atomic.cas, types.Array, types.Tuple, types.Any, types.Any)
+@lower(stubs.atomic.cas, types.Array, types.UniTuple, types.Any, types.Any)
+def ptx_atomic_cas(context, builder, sig, args):
```

Comment:
This function could probably be combined with the previous one, but I haven't figured out how to do this yet.

Reply:
Thanks, that is much more concise.

Reply:
Done.